### PR TITLE
Update azure yaml schema

### PR DESCRIPTION
**Pull request by haoozhang**
Update azure yaml schema to include more resource types and definitions.

### CODE REVIEW

**Review by saragluna — APPROVED** [review]

File: schemas/alpha/azure.yaml.json [hunk 1]
```diff
@@ -368,10 +368,14 @@
...
                         "enum": [
-                            "db.postgres",
                             "db.mysql",
+                            "db.postgres",
                             "db.redis",
                             "db.mongo",
+                            "db.cosmos",
+                            "messaging.servicebus",
+                            "messaging.eventhubs",
+                            "messaging.kafka",
```
Comment (rujche) [inline schemas/alpha/azure.yaml.json:378]: Add "storage" here. It should contain all types here:

> image

File: schemas/alpha/azure.yaml.json [hunk 2]
```diff
@@ -383,43 +387,19 @@
...
-                          "authType",
-                          "databaseName"
-                        ]
-                      }
-                    }
+                    { "if": { "properties": { "type": { "const": "db.mongo"  }}}, "then": { "$ref": "#/definitions/mongoDbResource"} },
+                    { "if": { "properties": { "type": { "const": "db.cosmos"  }}}, "then": { "$ref": "#/definitions/cosmosDbResource"} },
+                    { "if": { "properties": { "type": { "const": "messaging.servicebus"  }}}, "then": { "$ref": "#/definitions/serviceBusResource"} },
+                    { "if": { "properties": { "type": { "const": "messaging.eventhubs"  }}}, "then": { "$ref": "#/definitions/eventHubsResource"} },
+                    { "if": { "properties": { "type": { "const": "messaging.kafka"  }}}, "then": { "$ref": "#/definitions/kafkaResource"} }
```
Comment (rujche) [inline schemas/alpha/azure.yaml.json:402]: Same here. Add "storage"

File: schemas/alpha/azure.yaml.json [hunk 3]
```diff
@@ -1331,6 +1309,158 @@
                     }
                 }
             }
+        },
+        "sqlDbResource": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "type": true,
+                "uses": true,
```
Comment (rujche) [inline schemas/alpha/azure.yaml.json:1318]: What does `"type": true` and `"uses": true` mean?

File: schemas/alpha/azure.yaml.json [hunk 4]
```diff
@@ -1331,6 +1309,158 @@
...
+                    "description": "The Azure resource supports this kind of authentication for connection.",
+                    "enum": [
+                        "USER_ASSIGNED_MANAGED_IDENTITY",
+                        "PASSWORD"
+                    ]
+                },
+                "databaseName": {
+                    "type": "string",
+                    "title": "The name of Azure resource",
+                    "description": "The name of Azure resource."
```
Comment (rujche) [inline schemas/alpha/azure.yaml.json:1331]: Not `the name of azure resource`. Please refer to this PR: https://github.com/azure-javaee/azure-dev/pull/37/files

Comment (rujche) [reply schemas/alpha/azure.yaml.json:1331]: Both title and description should be updated.

File: schemas/alpha/azure.yaml.json [hunk 5]
```diff
@@ -1331,6 +1309,158 @@
...
+        "mongoDbResource": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "type": true,
+                "uses": true,
+                "databaseName": {
+                    "type": "string",
+                    "title": "The name of Azure resource",
+                    "description": "The name of Azure resource."
```
Comment (rujche) [inline schemas/alpha/azure.yaml.json:1344]: Same here.

File: schemas/alpha/azure.yaml.json [hunk 6]
```diff
@@ -1331,6 +1309,158 @@
...
+        "cosmosDbResource": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "type": true,
+                "uses": true,
+                "databaseName": {
+                    "type": "string",
+                    "title": "The name of Azure Cosmos DB resource",
+                    "description": "The name of Azure resource."
```
Comment (rujche) [inline schemas/alpha/azure.yaml.json:1357]: Same here.

File: schemas/alpha/azure.yaml.json (hunk 2 above)
Comment (haoozhang) [reply schemas/alpha/azure.yaml.json]: Added.

File: schemas/alpha/azure.yaml.json (hunk 3 above)
Comment (haoozhang) [reply schemas/alpha/azure.yaml.json:1322]: Similar as above definitions, I think they indicate that the condition is always valid here (skip validation), and these two properties should follow the validation rules defined earlier.

File: schemas/alpha/azure.yaml.json (hunk 4 above)
Comment (haoozhang) [reply schemas/alpha/azure.yaml.json]: Updated.

File: schemas/alpha/azure.yaml.json (hunk 5 above)
Comment (haoozhang) [reply schemas/alpha/azure.yaml.json]: Updated.

File: schemas/alpha/azure.yaml.json (hunk 6 above)
Comment (haoozhang) [reply schemas/alpha/azure.yaml.json]: Updated.

File: schemas/alpha/azure.yaml.json [hunk 7]
```diff
@@ -1240,9 +1222,8 @@
                     "title": "Type of resource",
                     "description": "The type of resource to be created. (Example: db.postgres)",
                     "enum": [
-                        "db.postgres",
```
Comment (rujche) [inline schemas/alpha/azure.yaml.json:1243]: The description should be updated. Don't use db.postgres as example.

**Review by rujche — CHANGES_REQUESTED** [review]

File: schemas/alpha/azure.yaml.json (hunk 7 above)
Comment (haoozhang) [reply schemas/alpha/azure.yaml.json:1243]: Updated.

**Review by rujche — APPROVED** [review]
LGTM.

**Pull request merged**